test/values/dynamic-value.js

```diff
@@ -0,0 +1,98 @@
+import setup from '../helper';
+
+import { DynamicValue, TextElement, TextValue, CheckboxElement } from '../../src';
+
+let record = null;
+let checklist = null;
+
+beforeEach((done) => {
+  setup((rec) => {
+    record = rec;
+    checklist = record.formValues.find('checklist');
+    done();
+  });
+});
+
+describe('DynamicValue', () => {
+  describe('isEmpty', () => {
+    it('is true when there are no items', () => {
+      checklist = new DynamicValue('element', []);
+      checklist.isEmpty.should.eql(true);
+    });
+
+    it('is false when there are items', () => {
+      checklist.isEmpty.should.eql(false);
+    });
+  });
+
+  describe('addItem', () => {
+    it('adds an item', () => {
+      const newItem = checklist.createNewItem();
+      const newElement = new TextElement('element', {key: '2', data_name: '2'});
+      newItem.elements.push(newElement);
+      const newValue = new TextValue(newElement, 'testtext');
+      newItem.values.set('2', newValue);
+
+      newItem.elements[0].should.be.instanceof(TextElement);
+      newItem.elements.length.should.eql(1);
+
+      checklist.insertItem(newItem);
+
+      checklist.itemIndex(newItem.metadata.id).should.eql(1);
+      checklist.items.length.should.eql(2);
+      const testElement = checklist.items[1].elements[0];
+      testElement.should.be.instanceof(TextElement);
+      const testValue = checklist.items[1].values.get(testElement.key);
+      testValue.should.be.instanceof(TextValue);
+
+      checklist.items[1].toJSON().metadata.id.should.eql(newItem.metadata.id);
+      // checklist.items[1].toJSON().should.eql({ metadata: { id: 'aaa'}, elements: [], values: {} });
```

Comment:
was this still needed?

Reply:
This one is pending the ticket to turn the elements into JSON.